src/python_testing/TC_MOD_1_2.py

```diff
@@ -0,0 +1,108 @@
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+#
+# === BEGIN CI TEST ARGUMENTS ===
+# test-runner-runs:
+#   run1:
+#     app: ${ALL_CLUSTERS_APP}
+#     app-args: --discriminator 1234 --KVS kvs1 --trace-to json:${TRACE_APP}.json
+#     script-args: >
+#       --storage-path admin_storage.json
+#       --commissioning-method on-network
+#       --discriminator 1234
+#       --passcode 20202021
+#       --trace-to json:${TRACE_TEST_JSON}.json
+#       --trace-to perfetto:${TRACE_TEST_PERFETTO}.perfetto
+#       --PICS src/app/tests/suites/certification/ci-pics-values
+#       --endpoint 1
+#     factory-reset: true
+#     quiet: true
+# === END CI TEST ARGUMENTS ===
+
+import chip.clusters as Clusters
+from chip.testing.matter_testing import MatterBaseTest, TestStep, async_test_body, default_matter_test_main
+from chip.tlv import uint
+from mobly import asserts
+
+
+class TC_MOD_1_2(MatterBaseTest):
+
+    def steps_TC_MOD_1_2(self) -> list[TestStep]:
+        return [
+            TestStep(1, "TH reads the SupportedModes attribute from DUT", is_commissioning=True),
+            TestStep(2, "TH reads the CurrentMode attribute from the DUT",
+                     "Verify that the DUT response is an integer that is in the list of modes returned in step 1"),
+            TestStep(3, "TH reads the OnMode attribute from the DUT",
+                     "Verify that the DUT response is an integer that is in the list of modes returned in step 1"),
+            TestStep(4, "TH reads the StartUpMode attribute from the DUT",
+                     "Verify that the DUT response is an integer that is in the list of modes returned in step 1"),
+            TestStep(5, "TH reads the Description attribute from the DUT",
+                     "Verify that the DUT response provides human readable text string that describes the purpose of the mode select server."),
+            TestStep(6, "TH reads the StandardNamespace attribute from the DUT",
+                     "Verify that the DUT response provides a 16 bit enum (null is also acceptable)."),
+        ]
+
+    def pics_TC_MOD_1_2(self) -> list[str]:
+        return ["MOD.S"]
+
+    @async_test_body
+    async def test_TC_MOD_1_2(self):
+        self.step(1)
+        # Establishing TH1
+        self.endpoint = self.get_endpoint()
+        self.th1 = self.default_controller
+        MOD_cluster = Clusters.ModeSelect
+        attributes = MOD_cluster.Attributes
+
+        if await self.attribute_guard(endpoint=self.endpoint, attribute=attributes.SupportedModes):
+            mode_options = await self.read_single_attribute(dev_ctrl=self.th1, node_id=self.dut_node_id, endpoint=self.endpoint, attribute=attributes.SupportedModes)
```

Comment:
The checks in the test plan are more in depth here. I don't think we have a machine-readable version of the standard namespaces right now, so leave that piece for now, but there are some checks that should be in here. See test plan. In particular, unique modes.